zetaclient/chains/solana/observer/observer.go

```diff
@@ -133,3 +145,24 @@
 
 	return nil
 }
+
+// SetTxResult sets the tx result for the given nonce
+func (ob *Observer) SetTxResult(nonce uint64, result *rpc.GetTransactionResult) {
+	ob.Mu().Lock()
+	defer ob.Mu().Unlock()
+	ob.finalizedTxResults[ob.GetTxID(nonce)] = result
+}
+
+// GetTxResult returns the tx result for the given nonce
+func (ob *Observer) GetTxResult(nonce uint64) *rpc.GetTransactionResult {
+	ob.Mu().Lock()
+	defer ob.Mu().Unlock()
+	return ob.finalizedTxResults[ob.GetTxID(nonce)]
+}
```

Comment:
Do you plan to use DB for this? Otherwise we need to have eviction policies for this map. Also consider using `(*rpc.GetTransactionResult, bool) // found` for results

Reply:
@swift1337 I created a issue to track this https://github.com/zeta-chain/node/issues/2608. Can you confirm is the issue what you meant for eviction?

For the `(*rpc.GetTransactionResult, bool)` return type, I wanted to keep it unchanged to make the code simplified. Return an additional `found` would require the user (of the map) code to check both `found` flag and `nil` pointer. Current code requires the user to only check `nil` pointer.

Reply:
I mean, do you have a code that occasionally does smth like `delete(ob.finalizedTxResults, txID)`?

Reply:
No we delete it nowhere. We'll need to figure a reliable way to tell WHEN/WHICH can be deleted before the memory footprint becomes a real problem in future. For instance, running zetaclient for a year without break or having huge traffic. I'm eager to see memory footprint issue happen.